api/kv.go

```diff
@@ -0,0 +1,300 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+// A KVClient is used to perform reads and writes against a KV secrets engine in Vault.
+//
+// The mount path is the location where the target KV secrets engine resides
+// in Vault. The version refers to the version of the target KV secrets engine.
+//
+// Vault development servers tend to have "secret" as the mount path
+// and 2 as the version, as these are the default settings when a server
+// is started in -dev mode.
```

Comment:
It may seem strange that I have this note here instead of actually providing these as default values for the client. My team considered that for awhile, but came to the conclusion that being explicit here might be better, since v1 and v2 are both actively used for different use cases (v1 being more performant). 

Plus, default values in Go requires either variadic parameters (which is awkward when there will always be just one value for that param) or something like the functional options pattern, which once again seems like overkill for helper methods that are supposed to be immediately clear as to how to use them.

All that led to me just deciding to add this comment for clarity for folks who are super unfamiliar with Vault even at the dev server level.